### ISSUE TITLE
Thoughts and concerns about a future proposed installer.

### ISSUE THREAD

**Issue by Maikuolan** [CIDRAM/CIDRAM#27]
*Thoughts and concerns about a future proposed installer: Just creating this issue to share some thoughts. This issue doesn't describe any specific plans, intentions, promises, bugs, etc. Related thoughts and opinions from others is invited.*

__Problems.__
- Due to the numerous different software packages that exist to which CIDRAM might be attached, and thus, the difficulty in adequately and reliably predicting the environment to which it might be installed in regards to the needs of any one specific user, the task of establishing "hooks" between the target software package and CIDRAM upon installation in any automatic capacity (i.e., by way of an installer, installation wizard, etc) is one which may be difficult to complete in an adequately reliable way, and not accounting for any potential failure to install CIDRAM, may also be potentially dangerous in some circumstances, such as if a "hook" is incorrectly inserted, causing the target software package to break. Due to similar and shared goals, this problem relates to both CIDRAM and phpMussel.
- Creating various plugins and extensions for CIDRAM and phpMussel is a solution which has been considered for some time, and in many cases, serves as a viable means to ensure safe, automatic installation for the specific software packages which these plugins and extensions are built for. Of course, it is limited to those specific software packages (i.e., providing no benefit for unrelated software packages), and an issue unrelated to that of building any manner of generic installer for CIDRAM and phpMussel.
- Assuming that the insertion of "hooks" remains a manual step in the installation process, the installation process can otherwise be mostly automated, but the benefit of building an installer for this as opposed to continuing to rely on the currently available installation methods becomes questionable, I think, in that there isn't really all that many differences remaining anyhow, aside from the possibility of some future fancy GUI or the likes.
- In the event of inadequate CHMOD permissions or other related restrictions, manual installation may be enforced in many cases anyhow, or at the very least, ensuring adequate CHMOD permissions may be another step which must remain manual in some cases.
- At this point, I'm having doubts about the necessity of building an installer at all, and having some difficult conceptualising what form it may take.

__Benefits.__
- People like installers. In many cases, people expect them to exist. They can create a sense of ease that it should be an easy task to get some piece of software up and running in an easy and timely manner. It can make things *look* a little nicer. Certainly, in many cases (e.g., when some piece of software relies on interacting with database connectors, need to install registration data, etc), installers do make things much easier. Not sure any of that applies in this case though; CIDRAM and phpMussel handle everything using their own files.
- It's something which has been mentioned in the documentation for a very long time, and most users at this point will assume that I'm going to be building an installer at some point.

**Comment by DanielRuf** [comment]
I am unsure about the benefits. We have composer and one line to hook it into the PHP process.

Or what do you mean with an installer?

**Comment by Maikuolan** [comment]
I'm thinking the same thing, and thinking about dropping *"build a front-end installer"* from the to-do list.

Originally, when the project was first started, I think there were some good reasons to include an installer: As a way to automatically regenerate the configuration when changes are made, as a way to readily fetch the latest project files when required to install and/or update, etc. However, since then, we've implemented support for composer, which is already capable of fetching any necessary files, installing everything, updating and such, and we've implemented the front-end management, which in particular, includes the configuration page, which is capable of making changes to the configuration and regenerating it as per when necessary. All the original reasons for including an installer don't really exist anymore, due to the various other changes that've happened since then which already take care of all these things.

For projects with a higher level of complexity and a higher number of dependencies, requirements and such things, having an installer to do everything can be very useful, but in our case.. There isn't much complexity involved anyway, and it's already very easy and simple to install everything.

In short.. An idea that seemed good in the past, but which kind of seems pointless now. Just wondering whether I might be missing anything here, or whether there are other considerations that I've forgotten to take into account, whether it's something that might still be on interest for whatever reason and such.

**Comment by Maikuolan** [comment]
Yeah.. Scratched from the to-do list.
